tools/cibadmin.c

```diff
@@ -483,28 +478,33 @@ main(int argc, char **argv)
                 dangerous_cmd = TRUE;
                 break;
             case 'S':
-                if (optarg != NULL) {
-                    if (!strcmp(optarg, "auto")) {
-                        acl_eval_how = acl_eval_auto;
-                    } else if (!strcmp(optarg, "namespace")) {
-                        acl_eval_how = acl_eval_namespace;
-                    } else if (!strcmp(optarg, "text")) {
-                        acl_eval_how = acl_eval_text;
-                    } else if (!strcmp(optarg, "color")) {
-                        acl_eval_how = acl_eval_color;
-                    } else {
-                        fprintf(stderr, "Unrecognized value for --show-access: \"%s\"\n",
-                               optarg);
-                        ++argerr;
-                    }
+                if (pcmk__str_eq(optarg, "auto", pcmk__str_null_matches)) {
+                    acl_render_mode = pcmk__acl_render_default;
+
+                } else if (strcmp(optarg, "namespace") == 0) {
+                    acl_render_mode = pcmk__acl_render_namespace;
+
+                } else if (strcmp(optarg, "text") == 0) {
+                    acl_render_mode = pcmk__acl_render_text;
+
+                } else if (strcmp(optarg, "color") == 0) {
+                    acl_render_mode = pcmk__acl_render_color;
+
                 } else {
-                    acl_eval_how = acl_eval_auto;
+                    fprintf(stderr,
+                            "Unrecognized value for --show-access: '%s'\n",
+                            optarg);
+                    ++argerr;
                 }
-                /* XXX this is a workaround until we unify happy paths for
-                       both a/sync handling; the respective extra code is
-                       only in sync path now, but does it matter at all for
-                       query-like request wrt. what blackbox users observe? */
-                command_options |= cib_sync_call;
+
+                /* The ACL render modes work only with sync calls due to
+                 * differences in output handling. It's unlikely to matter to
+                 * the user whether the call is sync or async; it's an
+                 * implementation detail that shouldn't have a noticeable
+                 * effect, especially for queries.
+                 */
```

Comment:
Looking at this again, I misled you :)

sync isn't an advanced option, it really does what it says -- without it, cibadmin just submits the request and returns without waiting for confirmation of completion. So what the comment was originally trying to say is that for a CIB query, async is irrelevant, since it has to wait for the result in any case to display it.